test/FastTests/Sparrow/Hashing.cs

```diff
@@ -387,114 +284,250 @@ public unsafe void XXHash32_StreamedHashingEquivalence(int bufferSize)
             int iteration = 1;
             do
             {
-                blockSize = Hashing.Streamed.XXHash32.Alignment * iteration;
+                blockSize = Hashing.Streamed.XXHash64.Alignment * iteration;
 
-                var context = new Hashing.Streamed.XXHash32Context {Seed = seed};
-                Hashing.Streamed.XXHash32.BeginProcess(ref context);
+                var context = new Hashing.Streamed.XXHash64Context {Seed = seed};
+                Hashing.Streamed.XXHash64.Begin(ref context);
                 fixed (byte* buffer = values)
                 {
                     byte* current = buffer;
                     byte* bEnd = buffer + bufferSize;
                     do
                     {
                         int block = Math.Min(blockSize, (int)(bEnd - current));
-                        Hashing.Streamed.XXHash32.Process(ref context, current, block);
+                        Hashing.Streamed.XXHash64.Process(ref context, current, block);
                         current += block;
                     }
                     while (current < bEnd);
                 }
 
                 iteration++;
 
-                var result = Hashing.Streamed.XXHash32.EndProcess(ref context);
-                var expected = Hashing.XXHash32.Calculate(values, -1, seed);
+                var result = Hashing.Streamed.XXHash64.End(ref context);
+                var expected = Hashing.XXHash64.Calculate(values, -1, seed);
 
                 Assert.Equal(expected, result);
             }
             while (blockSize <= bufferSize);
         }
 
-        [Theory]
-        [MemberData("BufferSize")]
-        public unsafe void XXHash64_StreamedHashingEquivalence(int bufferSize)
+        [Fact]
+        public unsafe void XXHash64_HashingEquivalenceWithReference()
         {
             var rnd = new Random(1000);
 
-            byte[] values = new byte[bufferSize];
+            byte[] values = new byte[1024];
             rnd.NextBytes(values);
 
-            uint seed = 233;
+            fixed (byte* valuePtr = values)
+            {
+                for (int i = 1; i < values.Length; i++)
+                {
+                    var expected = XXHash64Reference(valuePtr, (ulong)i, seed: 1337);
+                    var result = Hashing.XXHash64.CalculateInline(values.AsSpan().Slice(0, i), 1337);
 
-            int blockSize;
-            int iteration = 1;
-            do
+                    Assert.Equal(expected, result);
+                }
+            }
+        }
+
+        private static unsafe ulong XXHash64Reference(byte* buffer, ulong len, ulong seed = 0)
```

Comment:
This is the old implementation, right?

Reply:
Yes, this is the old implementation we use as reference.